src/platform_impl/linux/x11/window.rs

```diff
@@ -120,7 +120,18 @@ impl UnownedWindow {
         pl_attribs: PlatformSpecificWindowBuilderAttributes,
     ) -> Result<UnownedWindow, RootOsError> {
         let xconn = &event_loop.xconn;
-        let root = event_loop.root;
+        // root should be a type ffi::Window and it is finally c_ulong.
+        // cf. https://docs.rs/x11-dl/2.19.1/x11_dl/xlib/type.Window.html
+        let root = if let Some(id) = pl_attribs.parent_id {
+            #[cfg(target_pointer_width = "64")]
```

Comment:
Isn't a window guaranteed to be a 32 bit value? This pattern probably isn't exclusive to this PR, but I don't think the `try_*` items are necessary, since `as` will lead to no loss of precision.

Reply:
When I read x11 code little, I found that the window IDs sometimes are typedefed as a pointer (https://github.com/rust-windowing/winit/pull/2246#issuecomment-1175601618) so I don't think the window IDs is not guaranteed to be a 32-bit value.

Reply:
That's strange, because `libxcb` relies on XIDs being 32-bit, so any change that would make an XID longer than 32 bits would break `libxcb`. In addition, the X11 protocol [defines XIDs to be 32 bits](https://www.x.org/docs/XProtocol/proto.pdf) (see page 5).

Reply:
Thanks, I made a misunderstanding for it. So these `try_into()` are not neccesary for me too, I'll fix it.

Reply:
Fix that!

Reply:
> `libxcb` relies on XIDs being 32-bit, so any change that would make an XID longer than 32 bits would break `libxcb`

Agreed, however, `x11-dl` has had a seemingly broken pointer-width-reliant type alias for a while now (making their `XID=c_ulong`, which is 64-bit on 64-bit Linux) yet it somehow works for us.

 > Fix that!

I'm a bit confused now. Do you mean you've fixed it, or that you're going to fix it? And if you mean that you've fixed it, then I'm a bit confused as to why you're using `u64`s and `cfg(target_pointer_width = "64")` still.

Reply:
> I'm a bit confused now. 

Sorry for my rough English. I want to say "Fixed". And I agree your comment: there's no reason to use `cfg(...)`.

Reply:
I think it's fixed.

Reply:
Right, before this fix https://github.com/rust-windowing/winit/pull/2246/commits/62c9408f28782040d5de30b224bfdafbd543f795 that compiler conditional remains. Now that's removed so I think it's good.